File: java/src/main/java/com/codingdojo/solution/FizzBuzz.java
package com.codingdojo.solution;

import javax.print.PrintService;

public class FizzBuzz {

    public FizzBuzz() {
    }

    public void playFizzBuzz(int start, int end) {
        for (int i = start; i <= end; i++) {
            String result = getFizzBuzzValue(i);
            System.out.println(result);
        }
    }

    public String getFizzBuzzValue(int i) {
        boolean fizz = isFizz(i);
        boolean buzz = isBuzz(i);

        if (fizz && buzz) {
            return "FizzBuzz";
        } else if (fizz) {
            return "Fizz";
        } else if (buzz) {
            return "Buzz";
        } else {
            return String.valueOf(i);
        }
    }

    private boolean isFizz(int i) {
        return (i % 3 == 0) || containsDigit(i, 3);
    }

    private boolean isBuzz(int i) {
        return (i % 5 == 0) || containsDigit(i, 5);
    }

    private boolean containsDigit(int number, int digit) {
        return String.valueOf(number).contains(String.valueOf(digit));
    }
}

File: python/com/codingdojo/solution/fizz_buzz.py
from io import StringIO
import sys

# The FizzBuzz functions
def is_fizz(n):
    return n % 3 == 0 or '3' in str(n)

def is_buzz(n):
    return n % 5 == 0 or '5' in str(n)

# The FizzBuzz function
def get_fizz_buzz_value(n):
    if is_fizz(n) and is_buzz(n):
        return "FizzBuzz"
    elif is_fizz(n):
        return "Fizz"
    elif is_buzz(n):
        return "Buzz"
    else:
        return str(n)

def fizz_buzz(start, end):
    for i in range(start, end + 1):
        print(get_fizz_buzz_value(i))

def capture_output(start, end):
    captured_output = StringIO()
    sys.stdout = captured_output
    fizz_buzz(start, end)
    sys.stdout = sys.__stdout__
    return captured_output.getvalue()

